runtime/parachains/src/disputes.rs

```diff
@@ -717,31 +806,21 @@ impl<T: Config> Pallet<T> {
 	/// Handle sets of dispute statements corresponding to 0 or more candidates.
 	/// Returns a vector of freshly created disputes.
 	///
+	/// Assumes `statement_sets` were already de-duplicated.
```

Comment:
Do duplicates ever make any sense? And if not, could we make it so that `CheckedMultiDisputeStatementSet` gets a smart constructor that ensures this invariant?

Reply:
This would require the `CheckedMultiDisputeStatementSet` to become a full type rather than an alias. I think it might make sense to create another alias of `DedupSortedMultiDisputeStatementSets` it could be created as the output of `fn deduplicate_and_sort_..` :+1: and then be required by anything following, besides add a `CheckedMultiDisputeStatementSet` that assures sorting.